src/python/tests/core/bot/fuzzers/libFuzzer/libfuzzer_launcher_test.py

```diff
@@ -1804,6 +1817,348 @@ def test_fuzz_with_mutations_using_ml_rnn(self, mock_execute, *_):
           '/fake/inputs-disk/temp-1337/mutations',
       ]])
 
+  @mock.patch('bot.fuzzers.libFuzzer.launcher.add_recommended_dictionary',
+              lambda x, y, z: False)
+  @mock.patch('multiprocessing.cpu_count', return_value=1)
+  @mock.patch('sys.stdout', new_callable=test_utils.MockStdout)
+  def test_fuzz_with_dataflow_tracing(self, mock_stdout, *_):
+    """Tests fuzzing with dataflow tracing."""
+    #//self.mock.cpu_count.return_value = 1
+    self.mock.do_dataflow_tracing.return_value = True
+
+    self.fs.CreateDirectory('/fake/corpus_dir')
+    self.fs.CreateFile('/fake/testcase_basic')
+    self.fs.CreateDirectory('/fake/dfsan_build')
+    self.fs.CreateFile('/fake/dfsan_build/fake_fuzzer')
+
+    os.environ['FUZZ_CORPUS_DIR'] = '/fake/corpus_dir'
+    os.environ['DATAFLOW_BUILD_DIR'] = '/fake/dfsan_build'
+
+    new_units_added = 11
```

Comment:
Lines 1837-1848 are basically duplicated on lines (in original file, not patched): 1363, 1436, 1493, 1614, 1722, 1769, and once more in this patch on line 2026. Can the code be shared instead?

I realize this patch isn't doing anything wrong given the state of this file, but I think the file could be improved quite a bit.

Reply:
Agree, it's redundant, but these lines likely will go away as we re-invent the launcher interface. Not sure it's worth investing in refactoring at this point.